test/acceptance/stdin-by-platform-test.js

```diff
@@ -0,0 +1,139 @@
+          `);
+          expect(result.stderr).toBeFalsy();
+        });
+      });
+    });
+  }
+
+  if (process.platform === 'linux' || process.platform === 'darwin') {
+    describe('in unix bash', function () {
+      describe('command: `cat template.hbs | ember-template-lint --filename template.hbs`', function () {
+        it('has exit code 1 and reports errors to stdout', async function () {
+          let result = await execa(
+            'cat',
+            [
+              'template.hbs',
+              '|',
+              require.resolve('../../bin/ember-template-lint.js'),
+              '--filename',
+              'template.hbs',
+            ],
+            { shell: true, reject: false, cwd: project.path('.') }
+          );
+
+          expect(result.exitCode).toEqual(1);
+          expect(result.stdout).toMatchInlineSnapshot(`
+            "template.hbs
+              1:4  error  Non-translated string used  no-bare-strings
+              1:25  error  Non-translated string used  no-bare-strings
+
+            ✖ 2 problems (2 errors, 0 warnings)"
+          `);
+          expect(result.stderr).toBeFalsy();
+        });
+      });
+
+      describe('command: `cat template.hbs | ember-template-lint --filename template.hbs -`', function () {
+        it('has exit code 1 and reports errors to stdout', async function () {
+          let result = await execa(
+            'cat',
+            [
+              'template.hbs',
+              '|',
+              require.resolve('../../bin/ember-template-lint.js'),
+              '--filename',
+              'template.hbs',
+              '-',
+            ],
+            { shell: true, reject: false, cwd: project.path('.') }
+          );
+
+          expect(result.exitCode).toEqual(1);
+          expect(result.stdout).toMatchInlineSnapshot(`
+            "template.hbs
+              1:4  error  Non-translated string used  no-bare-strings
+              1:25  error  Non-translated string used  no-bare-strings
+
+            ✖ 2 problems (2 errors, 0 warnings)"
+          `);
+          expect(result.stderr).toBeFalsy();
+        });
+      });
+
+      describe('command: `ember-template-lint --filename template.hbs < template.hbs`', function () {
+        it('has exit code 1 and reports errors to stdout', async function () {
+          let result = await execa(
+            require.resolve('../../bin/ember-template-lint.js'),
+            ['--filename', 'template.hbs', '<', 'template.hbs'],
+            { shell: true, reject: false, cwd: project.path('.') }
+          );
+
+          expect(result.exitCode).toEqual(1);
+          expect(result.stdout).toMatchInlineSnapshot(`
+            "template.hbs
+              1:4  error  Non-translated string used  no-bare-strings
+              1:25  error  Non-translated string used  no-bare-strings
+
+            ✖ 2 problems (2 errors, 0 warnings)"
+          `);
+          expect(result.stderr).toBeFalsy();
+        });
+      });
```

Comment:
This test seems like _exactly_ the same as the one for `win32`, can we restructure this module so that the tests that run on all platforms are shared then we have the ones that only run on non-win32?

Something like:

```js
describe('ember-template-lint executable', function() {
  // this test

  if (process.platform !== 'win32') {
    describe('posix environments', function() {
      // the non-windows tests
    });
  }
});
```

Reply:
I've added a commit to consolidate test cases as per your suggestions.